### ISSUE TITLE
Kubeflow creating workspace volumes in different availability zone than instances, failing

### ISSUE THREAD

**Issue by areiner-novetta** [kubeflow/kubeflow#5967]
/kind bug

**What steps did you take and what happened:**
Kubeflow deployed on a `kops`-created cluster, in AWS.  Prior to this it had been in use, with little maintenance for months.

We needed to add an instance type to our cluster that is not available in the original (single) zone we deployed it to (us-east-1**c**).  I successfully used `kops` to create a second subnet (in us-east-1**b**), and a new instance group in that subnet, and added it to the cluster autoscaler.  

When creating a new notebook server with parameters that guarantee scaling up in that instance group, it fails immediately due to:

```pod didn't trigger scale-up ... 1 node(s) had volume node affinity conflict```

I confirmed that the EBS volume it provisions is in the original zone/subnet (us-east-1**c**), while the instance it provisions is from the new instance group in "us-east-1**b**".  If I remove the workspace volume (for a transient notebook server), it successfully creates the new instance in the new zone "us-east-1**b**".

tl;dr:

Provisioned:
Instance:  us-east-1**b**
EBS Volume: us-east-1**c**

**What did you expect to happen:**
I expected it to create the new volume in "us-east-1**b**" to match the deployment zone of the instance it is scaling up.

Expected:
Instance:  us-east-1**b**
EBS Volume: us-east-1**b**

**Anything else you would like to add:**
I kind of expected there to be an option somewhere for this, but couldn't find anything regarding it.  I'm wondering if Kubeflow even handles multi-zone properly.

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): v1beta1
- kfctl version: (use `kfctl version`):  N/A
- Kubernetes platform: (e.g. `minikube`):  `kops` & AWS
- Kubernetes version: (use `kubectl version`):  1.18.17
- OS (e.g. from `/etc/os-release`):  "CentOS Linux 7 (Core)"
